eZ/Publish/Core/FieldType/Tests/Url/Gateway/DoctrineStorageTest.php

```diff
@@ -194,6 +194,29 @@ public function testUnlinkUrl()
         $this->assertEquals($expected, $result);
     }
 
+    /**
+     * @covers \eZ\Publish\Core\FieldType\Url\UrlStorage\Gateway\DoctrineStorage::getUrlsFromUrlLink
+     */
+    public function testGetUrlsFromUrlLink()
+    {
+        $gateway = $this->getStorageGateway();
+
+        $urlIds = [];
+        $urlIds[] = $gateway->insertUrl('https://ibexa.co/example1');
+        $urlIds[] = $gateway->insertUrl('https://ibexa.co/example2');
+        $urlIds[] = $gateway->insertUrl('https://ibexa.co/example3');
+
+        $gateway->linkUrl($urlIds[0], 10, 1);
+        $gateway->linkUrl($urlIds[1], 10, 1);
+        $gateway->linkUrl($urlIds[1], 12, 2);
+        $gateway->linkUrl($urlIds[2], 14, 1);
+
+        self::assertEquals(['https://ibexa.co/example1' => true, 'https://ibexa.co/example2' => true], $gateway->getUrlsFromUrlLink(10, 1), 'Did not get expected urlS for field 10');
+        self::assertEquals(['https://ibexa.co/example2' => true], $gateway->getUrlsFromUrlLink(12, 2), 'Did not get expected url for field 12');
+        self::assertEquals(['https://ibexa.co/example3' => true], $gateway->getUrlsFromUrlLink(14, 1), 'Did not get expected url for field 14');
+        self::assertEquals([], $gateway->getUrlsFromUrlLink(15, 1), 'Expected no urls for field 15');
```

Comment:
The problem with these tests here is that they're executed on SQLite only, so they don't test anything useful.
You rather should add proper integration test.

At some point (when I have time (tm)) these tests here will be dropped, as they negatively impact performance of unit test suite.

How does the bug manifest itself for the end user / on API level?

Reply:
Right.. So I have changed the test into an integration test.

The bug doesn't manifest itself as a functional defect. It causes duplicated rows in in the ezurl_object_link table. This is wrong and causes extra storage needs and some performance decrease ( not very noticeable in most cases though I )